### PR TITLE
Namespace as a function.

### PR DESCRIPTION
**Pull request by devel-pa**
For issue #33 found that is much easier to change a bit the plugin.
Now, the namespace property can be a string or a function.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -47,8 +52,13 @@ HandlebarsCompiler.prototype.compile = function(data, path, callback) {
   var error, key, ns, result, source;
   try {
     source = "Handlebars.template(" + (handlebars.precompile(data)) + ")";
-    if (this.namespace) {
+
+    if(this.isFunction(this.namespace))
```
Comment (es128) [inline index.js]: `typeof this.namespace === 'function'` should be sufficient

and please match code style: `if (cond)` with whitespace, and use brackets for multi-line `if`s

**Comment by devel-pa** [comment]
Thanks, I'll do it.
Also a question:
why not instead of
`"if (typeof " + ns + " === 'undefined'){ " + ns + " = {} }; "`
something like
`"ns = ns || {};"`
Thanks,
Paul

**Comment by es128** [comment]
I don't know, seems like it should work. Give it a try.

But it would be `ns + " = " + ns + " || {};"`

**Comment by devel-pa** [comment]
Indeed. I was lazy.
Still remains the issue of a namespace with multiple segments.

**Comment by es128** [comment]
Do you intend to address that in this PR? Perhaps just replace the slashes with underscores.

**Comment by devel-pa** [comment]
No, I don't want to address it in this PR.
Thanks

**Comment by es128** [comment]
Did you try out this latest code to ensure it behaves as expected (especially the client-side output)?

It really would be best if you could add a test for namespace setting as a function.

**Comment by devel-pa** [comment]
Tested on my project. Works as expected.

**Comment by es128** [comment]
Great, but still need to add a unit test to the PR to ensure the new function option doesn't get messed up in a future regression.

**Comment by devel-pa** [comment]
Sorry, you're right.
Anyway, the last commit is not passing the test. Works in browser but not in node. 
I'll revert and make the test for the function.

**Comment by es128** [comment]
Not sure what issue you observed in node, but perhaps declaring the namespace `var` before your refactor will fix it (and be cleaner regardless).

**Comment by devel-pa** [comment]
OK, it is working. I hope no other tool will complain abut it.

**Comment by es128** [comment]
so do this: https://github.com/brunch/handlebars-brunch/pull/34#issuecomment-71036026

**Comment by devel-pa** [comment]
Better without it for the moment. I'll find a solution later.

**Comment by es128** [comment]
the original code still works when there are slashes in the namespace?

**Comment by devel-pa** [comment]
The original is not working with slashes in namespace name, just in the path of the file as array.
The same with the function.

**Comment by es128** [comment]
yes that's what I meant... maybe I'm misunderstanding what you mean about 2 segments

**Comment by devel-pa** [comment]
JST.Templates

**Comment by es128** [comment]
```js
ns = ns.replace(/\W/, '_')
```

**Comment by devel-pa** [comment]
yes, i know, but is no more a namespace

**Comment by es128** [comment]
ok gotcha

so do you consider this PR ready now?

**Comment by devel-pa** [comment]
yes

**Comment by devel-pa** [comment]
I think I found a solution for full namespacing, but we will talk about it after this PR is accepted or rejected.

**Pull request merged**